### DISCUSSION TITLE
WASM CD cross compilation troubles

### DISCUSSION THREAD

**Discussion by QueenOfSquiggles** [bevyengine/bevy#14111] [Q&A] [answered]
Heyo, I'm trying to go off of the [official CI/CD example for bevy](https://github.com/bevyengine/bevy_github_ci_template) but I keep running into errors that I'm not sure how to handle.

The root of my problem seems to be:
```
warning: glib-sys@0.15.10: pkg-config has not been configured to support cross-compilation.
```
It recommends:
```
  Install a sysroot for the target platform and configure it via
  PKG_CONFIG_SYSROOT_DIR and PKG_CONFIG_PATH, or install a
  cross-compiling wrapper for pkg-config and set it via
  PKG_CONFIG environment variable.
```

And when I try that, I get errors while compiling `bevy_core_pipeline`

```
error[E0432]: unresolved imports `libc::c_char`, `libc::c_int`, `libc::c_ushort`, `libc::c_void`
 --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/glib-sys-0.15.10/src/manual.rs:4:12
```

When I try to do a web search for this I get all kinds of advice for different things like `cairo` and such. Hoping this might be something a more experienced bevy user might have experienced. 

Also for context:
- my github action file https://gist.github.com/QueenOfSquiggles/b56d4fce868558dfb371fa42507ad06a
- my `release-wasm` log for that action file: https://gist.github.com/QueenOfSquiggles/9e2b8056dd3d058fb3d7042c9cfad314
- my repo all this is happening in is here: https://github.com/QueenOfSquiggles/watery-wails

I'm trying to figure out if this is something to do with bevy (since AFAIK glib is a library for interacting with gtk windows) or if it's one of the crates I added for extra features.

As it stands right now I feel my "project" is a decent minimal reproduction since I really haven't started adding any code. I wanted to make sure my CI/CD would work before getting into the dirt of it.

I'm relatively new to Rust and Bevy but not to development. Hoping I'm just being really dumb about something obvious. 

Also I added a segment in my action that generates a helpful version string that is injected into the rust code before compiling so the binary knows what version of the code it is. If that's interesting, I'm happy to submit a PR to the official example.

**Comment by QueenOfSquiggles** [comment]
I'm also finding this thread which seems relevant? https://github.com/gtk-rs/gtk-rs-core/issues/416

**Comment by mockersf** [comment, marked as answer]
One of your dependencies, `bevy-panic-handler` can't compile in Wasm. It's backed by https://crates.io/crates/msgbox which doesn't support Wasm

**Reply by QueenOfSquiggles** [reply]
> One of your dependencies, `bevy-panic-handler` can't compile in Wasm. It's backed by https://crates.io/crates/msgbox which doesn't support Wasm

Omg thanks a ton! I know it might be a bother, but would you mind explaining how you reached that? I wanna be able to solve this by myself in the future. ❤️

**Reply by mockersf** [reply]
well anything about panics is generally a bad idea in Wasm...

I also looked into the crate you reported as causing the issue, `glib-sys`, by running `cargo tree --invert glib-sys` to find how you depended on it, which also pointed to `bevy-panic-handler`

**Reply by QueenOfSquiggles** [reply]
> well anything about panics is generally a bad idea in Wasm...
> 
> I also looked into the crate you reported as causing the issue, `glib-sys`, by running `cargo tree --invert glib-sys` to find how you depended on it, which also pointed to `bevy-panic-handler`

Thanks a ton! I had found the tree command and was trying to grep the dependencies. But the invert flag looks really helpful!
